pkg/volume/csi/csi_mounter.go

```diff
@@ -392,12 +392,11 @@ func (c *csiMountMgr) supportsFSGroup(fsType string, fsGroup *int64, driverPolic
 		return false
 	}
 
-	accessModes := c.spec.PersistentVolume.Spec.AccessModes
-	if c.spec.PersistentVolume.Spec.AccessModes == nil {
+	if c.spec.PersistentVolume == nil || c.spec.PersistentVolume.Spec.AccessModes == nil {
```

Comment:
We might want to add a new log line for this for skipping due to it missing a PersistentVolume? - Mostly as an easier to skim in the future thing.

Reply:
Another way of working around this is to specify a `FSGroupPolicy` in `CSIDriver` object created for CSI driver (in this case intel pmem driver). That might be better solution -  because at the end of day, we do not want to look at PV access modes to determine fsgroup support policy, otherwise only root users can read/write this volume. 

So I think warning should be updated to reflect something like:
> mounter.SetupAt Warning: skipping fsgroup permission change, no access mode available. The volume may only be accessible to root users.

Reply:
That warning doesn't mention CSIDriver.FSGroupPolicy. Do I understand correctly that if that is set, then this code here will not be reached?

It might be worthwhile to add that to the message, because without it users and fellow developers will not know what to do when they see the warning.

Reply:
Can CSIDriver.FSGroupPolicy really work for CSI ephemeral inline volumes? I'm lazy (sorry!) and did not look up the corresponding KEP, I'm just suspicious that NodePublishVolume (the only gRCP call for those kind of volumes) won't have the necessary information for a CSI driver to do the ownership changes.

Reply:
The ownership/permission change is performed by kubelet and should work for CSI ephemeral inline volumes - as long as CSI driver declares support for it via `CSIDriver.Spec.FSGroupPolicy` field.  It is performed *after* `NodePublish` and hence should behave same for regular CSI volumes and inline ephemeral volumes.

Reply:
> That warning doesn't mention CSIDriver.FSGroupPolicy. Do I understand correctly that if that is set, then this code here will not be reached?

Yes I think so. I have not personally tested it but code is pretty straightforward.

> It might be worthwhile to add that to the message, because without it users and fellow developers will not know what to do when they see the warning.

That is a good point and I was thinking about it but may be we should do a general sweep of such messages and figure out, how best to inform our users. My point is - a normal user probably can't even see kubelet logs and hence it is not super useful. I also did not wanted to block this PR while we decide how best to improve the messaging.

Reply:
Filed - https://github.com/kubernetes/kubernetes/issues/102738